src/alarms/notifications/GroupAndTemplateSelect.js

```diff
@@ -47,12 +47,24 @@ class GroupAndTemplateSelect extends Component {
       }
     );
   }
+  getMessageName(messages, messageId) {
+    if (messages.length === 0) return null;
+    const selectedMessage = messages.find(message => message.id === messageId);
+    return selectedMessage.name;
+  }
+  getGroupName(groups, groupId) {
+    if (groups.length === 0) return null;
+    const selectedGroup = groups.find(group => group.id === groupId);
+    return selectedGroup.name;
+  }
```

Comment:
For both these: handle the case that the message isn't found, even if you know it will be (this is called "defensive programming").

Reply:
Nice! Thanks for pointing it out. I now also handle the case if message isn't found.